### PR TITLE
Remove obsolete management components

### LINKED ISSUE #3887: Old tkr-source-controller needs to be removed when upgrading to (u)TKG 2.0
**Bug description**

In TKG 1.6 and prior, tkr-source-controller is installed via a ClusterResourceSet (with `ApplyOnce` mode). TKG 2.0 introduces tkr-service with broader functionality, so the old tkr-source-controller needs to be removed.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- (X) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

After upgrading from TKG 1.6 to uTKG, `tkr-controller-manager` deployment no longer exists in the `tkr-system` namespace.

**Steps to reproduce the bug**

1. Upgrade from TKG 1.6 to uTKG.
2. Notice that the `tkr-controller-manager` deployment still exists in the `tkr-system` namespace.

**Version** (include the SHA if the version is not obvious)

TKG 1.6

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by imikushin**
Remove obsolete management components when upgrading the management cluster. Currently, it's the old tkr-controller-manager deployment in the tkr-system namespace.

Signed-off-by: Ivan Mikushin <imikushin@vmware.com>

### Which issue(s) this PR fixes

Fixes #3887 

### Describe testing done for PR
Added a unit test.

### Release note

```release-note
When upgrading the management cluster to new TKG version, obsolete management components (currently, the tkr-controller-manager deployment in the tkr-system namespace) will be deleted prior to installing the new management packages.
```

### CODE REVIEW

File: tkg/client/management_components.go [hunk 1]
```diff
@@ -47,8 +48,58 @@ func (c *TkgClient) InstallOrUpgradeKappController(kubeconfig, kubecontext strin
...
+			Version: "v1beta1",
+			Kind:    "ClusterResourceSet",
+		}: {client.InNamespace(constants.TkrNamespace)},
+		{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    "Deployment",
+		}: {client.InNamespace(constants.TkrNamespace)},
+	}
+
```
Comment (12345lcr) [inline tkg/client/management_components.go:84]: Are these all we needed to cleanup? During my manual tests it seems that the new tkr won't be synced unless
* Delete tkr-<tkr version> configmap
* Delete all old tkrs

Comment (imikushin) [reply tkg/client/management_components.go:84]: Hey Chenrui, there's no need to delete old TKRs. They should probably stay, as they might have been used to create (legacy-style) clusters.

Any new TKRs will be created from TKR packages fetched by the new TKR Source Controller.

Also, TKR BOM ConfigMaps will be saved in `tog-system` namespace now, as well as replicated to `tkr-system` for the sake of backwards compatibility. See #3753

**Review by 12345lcr — APPROVED** [review]

**Review by prkalle — COMMENTED** [review]
LGTM.  Thanks

**Review by tenczar — APPROVED** [review]
LGTM

**Review by wjun — APPROVED** [review]
LGTM

**Pull request merged**